src/mersenne.ts

```diff
@@ -0,0 +1,38 @@
+const Gen = require('../vendor/mersenne').MersenneTwister19937;
+
+export class Mersenne {
+  private gen = new Gen();
+
+  constructor() {
+    this.gen.init_genrand(new Date().getTime() % 1000000000);
+  }
+
+  rand(max?: number, min?: number) {
+    // TODO @Shinigami92 2022-01-11: This is buggy, cause if min is not passed but only max,
+    // then min will be undefined and this result in NaN for the whole function
+    if (max === undefined) {
+      min = 0;
+      max = 32768;
+    }
```

Comment:
I think your code needs to check if min is `undefined` as well.
for example called with `rand(undefined, 20)`
the min will be overridden to 0

my suggest is Optional Parameters
https://flexiple.com/optional-parameter-javascript/
```suggestion
  rand(max = 32768, min = 0) {
    // TODO @Shinigami92 2022-01-11: This is buggy, cause if min is not passed but only max,
    // then min will be undefined and this result in NaN for the whole function
```

Reply:
yeah, we will open PRs for each of these issues individually after the compile time migration to TS